src/library-aux/scala/Singleton.scala

```diff
@@ -0,0 +1,17 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2002-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package scala
+
+/** Scala has concept of object.An object is a class that has exactly one instance. It is created lazily.
+ *
+ *  A singleton type is of the form p.type, where p is a path pointing to a value expected to conform to scala.AnyRef. 
+ *  The type denotes the set of values consisting of null and the value denoted by p.
```

Comment:
you should use Scaladoc syntax so that code snippets like `p.type`, `p`, and `scala.AnyRef` are rendered in a monospaced font.

Reply:
this looks like spec-ese, was it copied from the spec? it's fine to copy, but the text doesn't seem completely apropos here.

what's missing here is any idea of what the `Singleton` type itself is actually for. it's fine to explain `object` and explain singleton types, but it doesn't get at the heart of the matter. the heart of the matter is the type `scala.Singleton`. what is it, what does it do, what is it for...?

Reply:
Working on it. Need to understand usage of scala.Singleton.